File: expo/sample-ui/src/components/use-smile-id-sample-confirmation.ts
import { Alert } from 'react-native';

/// The platform's own alert, asking before an action that deletes something; the confirm action is destructive.
export const smileIDSampleConfirm = ({
  title,
  message,
  confirmLabel,
  onConfirm,
}: {
  readonly title: string;
  readonly message: string;
  readonly confirmLabel: string;
  readonly onConfirm: () => void;
}): void =>
  Alert.alert(title, message, [
    { text: 'Cancel', style: 'cancel' },
    { text: confirmLabel, style: 'destructive', onPress: onConfirm },
  ]);

File: ios/SampleUI/Sources/SampleUI/Components/UseSmileIDSampleConfirmation.swift
import SwiftUI

public extension View {
  /// The platform's own alert, asking before an action that deletes something; the confirm action is destructive.
  func useSmileIDSampleConfirmation(
    isPresented: Binding<Bool>,
    title: String,
    message: String,
    confirmLabel: String,
    confirmTestId: String,
    onConfirm: @escaping () -> Void
  ) -> some View {
    alert(title, isPresented: isPresented) {
      Button(confirmLabel, role: .destructive, action: onConfirm)
        .accessibilityIdentifier(confirmTestId)
      Button("Cancel", role: .cancel) {}
    } message: {
      Text(message)
    }
  }
}
